### PR TITLE
Upgrade serverless

### PR DESCRIPTION
**Pull request by dpiatek**
Upgrade serverless to a modern version.

### CODE REVIEW

File: services/serverless.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
-service: "website-honestly-${env:ENVIRONMENT_NAME}"
+service: 'website-honestly-${env:ENVIRONMENT_NAME}'
```
Comment (dpiatek) [inline services/serverless.yaml]: This file has just `prettier` changes

File: webpack.base.config.js [hunk 2]
```diff
@@ -8,6 +8,8 @@ const publicPath = `/${process.env.URL_BASENAME || ''}`;
 const robots = process.env.ALLOW_ROBOTS ? 'robots-allow.txt' : 'robots-disallow.txt';
 
 const baseConfig = {
+  stats: 'minimal',
+
```
Comment (dpiatek) [inline webpack.base.config.js]: These minimize the webpack output in CI etc

File: services/publish.js [hunk 3]
```diff
@@ -18,6 +18,11 @@ export default function doPublish(_, __, cb) {
     .then(removeArchived(bucketName))
     .then(compileSite)
     .then(pages => Promise.all(pages.map(uploadPage)))
-    .then(data => cb(null, data))
+    .then(() =>
+      cb(null, {
+        statusCode: 200,
+        body: JSON.stringify({ results: 'Publish successful' }),
+      }),
+    )
```
Comment (dpiatek) [inline services/publish.js]: https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html#api-gateway-simple-proxy-for-lambda-output-format

**Comment by SiAshbery** [comment]
Merging as this upgrade is required to fix another issue. New tests will be added later.

**Pull request merged**